File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <unordered_set>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> seen;
        for (int i : nums) {
            if (seen.contains(i))
                return true;
            seen.insert(i);
        }  
        return false;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest = 0
        
        i = 0
        while i < len(s):
            seen = {}
            while i < len(s):
                if s[i] in seen:
                    i = seen[s[i]] + 1
                    break
                seen[s[i]] = i
                i += 1
            if len(seen) > longest:
                longest = len(seen)

        return longest